pkg/controller/cassandradatacenter/controller/statefulset.go

```diff
@@ -132,66 +52,44 @@ func CreateOrUpdateStatefulSet(
 	return statefulSet, err
 }
 
-func setControllerReference(
-	statefulSet *v1beta2.StatefulSet,
-	cdc *cassandraoperatorv1alpha1.CassandraDataCenter,
-	podSpec *corev1.PodSpec,
-	dataVolumeClaim *corev1.PersistentVolumeClaim,
-	reconciler *CassandraDataCenterReconciler,
-) error {
-
-	statefulSet.Spec = newStatefulSetSpec(cdc, podSpec, dataVolumeClaim)
-
-	if err := controllerutil.SetControllerReference(cdc, statefulSet, reconciler.scheme); err != nil {
-		return err
-	}
-
-	return nil
-}
-
-func newStatefulSetSpec(
-	cdc *cassandraoperatorv1alpha1.CassandraDataCenter,
-	podSpec *corev1.PodSpec,
-	dataVolumeClaim *corev1.PersistentVolumeClaim,
-) v1beta2.StatefulSetSpec {
 
+func newStatefulSetSpec(cdc *cassandraoperatorv1alpha1.CassandraDataCenter, podSpec *corev1.PodSpec, dataVolumeClaim *corev1.PersistentVolumeClaim) *v1beta2.StatefulSetSpec {
 	podLabels := DataCenterLabels(cdc)
 
-	return v1beta2.StatefulSetSpec{
-		ServiceName: "cassandra",
+	return &v1beta2.StatefulSetSpec{
+		ServiceName: "cassandra", // TODO: correct service name? this service should already exist (apparently)
 		Replicas:    &cdc.Spec.Nodes,
 		Selector:    &metav1.LabelSelector{MatchLabels: podLabels},
 		Template: corev1.PodTemplateSpec{
 			ObjectMeta: metav1.ObjectMeta{Labels: podLabels},
 			Spec:       *podSpec,
 		},
 		VolumeClaimTemplates: []corev1.PersistentVolumeClaim{*dataVolumeClaim},
+		PodManagementPolicy: v1beta2.OrderedReadyPodManagement,
 	}
 }
 
-func newPodSpec(
-	cdc *cassandraoperatorv1alpha1.CassandraDataCenter,
-	podInfoVolume *corev1.Volume,
-	containers []corev1.Container,
-) *corev1.PodSpec {
+func newPodSpec(cdc *cassandraoperatorv1alpha1.CassandraDataCenter, volumes []corev1.Volume, containers []corev1.Container) *corev1.PodSpec {
+	// TODO: any way to pass volumes and containers as pointers?
```

Comment:
As mentioned in this TODO, should we be passing slices-of-pointers for both volumes and containers? AFAIK there isn't an easy (or built-in) way to convert `[]*corev1.Volume` to `[]corev1.Volume` except a for loop.

Reply:
Actually, I'm just going to remove this function since it basically just wraps the creation of a very simple struct.

Reply:
I would leave it as is, we are using new... methods in createOrUpdateStatefulSet so why just use direct initialisation there? it is more consistent right now too, if we change this and it starts to be complicated we would move it into method anyway